procrustes/permutation.py

```diff
@@ -132,92 +132,69 @@ def permutation(
 
     # compute cost matrix C = A.T B
     c = np.dot(new_a.T, new_b)
-    # solve linear sum assignment problem to get the row/column indices of optimal assignment
-    row_ind, col_ind = linear_sum_assignment(c, maximize=True)
-    # make the permutation matric by setting the corresponding elements to 1
-    p = np.zeros(c.shape)
-    p[(row_ind, col_ind)] = 1
+    # compute permutation matrix using Hungarian algorithm
+    p = _compute_permutation_hungarian(c)
     # compute one-sided permutation error
     error = compute_error(new_a, new_b, p)
 
     return ProcrustesResult(new_a=new_a, new_b=new_b, t=p, error=error)
 
 
 def permutation_2sided(
-        a,
-        b,
-        single=True,
-        pad=False,
-        unpad_col=False,
-        unpad_row=False,
-        translate=False,
-        scale=False,
-        check_finite=True,
-        mode="normal1",
-        iteration=500,
-        tol=1.0e-8,
-        kopt=None,
-        weight=None,
-        lapack_driver="gesvd"
+    a,
+    b,
+    single=True,
+    method="kopt",
+    guess_p1=None,
+    guess_p2=None,
+    pad=False,
+    unpad_col=False,
+    unpad_row=False,
+    translate=False,
+    scale=False,
+    check_finite=True,
+    options=None,
+    weight=None,
+    lapack_driver="gesvd",
 ):
-    r"""Two-sided permutation Procrustes.
+    r"""Perform two-sided permutation Procrustes.
 
     Parameters
     ----------
     a : ndarray
-        The 2d-array :math:`\mathbf{A}_{m \times n}` which is going to be transformed.
+        The 2d-array :math:`\mathbf{A}` which is going to be transformed.
     b : ndarray
-        The 2d-array :math:`\mathbf{B}_{m \times n}` representing the reference.
+        The 2d-array :math:`\mathbf{B}` representing the reference matrix.
     single : bool
-        If true, the two permutation are assumed to be the same (i.e. two-sided permutation
-        Procrustes with one transformation). Further,
-        (1) if the input matrices `a` and `b` are symmetric within the threshold of 1.e-5,
-        undirected graph matching algorithm from [1] will be applied;
-        (2) if the input matrices `a` and `b` are not symmetric, the directed graph
-        matching algorithm from [1] is applied.
-        If false, the two permutation matrices can be different, i.e.
-        it is the two-sided permutation Procrustes with two transformations
-        (known as the regular two-sided permutation Procrustes here). An flip-flop
-        algorithm taken from [2] is used to iteratively solve this problem. Global optima is not
-        guaranteed here and may need to do an additional k-opt search.
-        Default=True.
+        If `True`, the single-transformation Procrustes is performed to obtain :math:`\mathbf{P}`.
+        If `False`, the two-transformations Procrustes is performed to obtain :math:`\mathbf{P}_1`
+        and :math:`\mathbf{P}_2`.
```

Comment:
We should add **optional** after single,

```python
single : bool, optional
    If `True`, the single-transformation Procrustes is performed to obtain :math:`\mathbf{P}`...
```

Reply:
done. thanks.